pkg/parser/format.go

```diff
@@ -0,0 +1,17 @@
+package parser
+
+import (
+	"strings"
+)
+
+// DetermineFileFormat determines the file format based on its extension
+func DetermineFileFormat(filename string) (format string) {
+	if strings.HasSuffix(filename, ".xml") {
+		return "rss"
+	} else if strings.HasSuffix(filename, ".json") {
+		return "json"
```

Comment:
If you decide that now it is `jsonp`. How do you know that you updated all the places where `jsonp` should be and did not touch the places where `json` should be...? There are a lot of hardcoded strings in your code...
<img width="254" alt="image" src="https://github.com/antonchaban/news-aggregator/assets/352913/070fe713-4575-4516-a180-4d065ed3c171">